niceday-api/index.test.js

```diff
@@ -1,5 +1,80 @@
-describe('Temporary Mock Test Suite', () => {
-  it('Mock Test Case', () => {
-    expect(true).toEqual(true);
+require('isomorphic-fetch');
+
+const NICEDAY_TEST_SERVERPORT = 8080;
+const NICEDAY_TEST_USER_ID = 38527;
+const MOCK_USER_DATA = {
+  id: NICEDAY_TEST_USER_ID,
+  userProfile: {
+    firstName: 'Mr Mock',
+  },
+};
+
+// Contains all tests which require a mocked Senseserver
+describe('Tests on niceday-api server using mocked goalie-js', () => {
+  let server;
+
+  beforeAll(() => {
+    // Mock the constructor of the SenseNetwork submodule of goaliejs to return
+    // a mock with the getContact() method, which itself returns a promise containing
+    // the MOCK_USER_DATA. SenseServer.Alpha needs to be mocked too since it is used
+    // in calls to the constructor.
+    //
+    // More comprehensive documentation can be found here:
+    // https://en.wikipedia.org/wiki/Lovecraftian_horror
```

Comment:
LOL

Reply:
To be honest I think this is quite sensible and not that complicated actually. Maybe we should refer to https://en.wikipedia.org/wiki/Supernatural_fiction or https://en.wikipedia.org/wiki/Weird_fiction instead?

Reply:
Haha, I think the true horror was the non-euclidean journey by which I arrived at this code. I'm just sure there is some cleaner way to present it than what I have created here. I want to keep the whole mock self-contained in one bit of the code though, so that we're not encouraged to make a mock more complicated than the code being tested...